### ISSUE TITLE
Clarify "three tests"?

### ISSUE THREAD

**Issue by ZukkyBaig** [neilstudd/technical-tests#5]
Hi all,

First of all, nice to meet you :), i'm Zukky Baig. I have a question I hope you don't mind answering. In the instructions to the technical test it states "write three tests that you think provide the best coverage of the functionality that is possible in 3 tests.". This can be interpreted in many ways, can you clarify? For instance, in Cucumber you can have a feature file with multiple scenario's, does a feature file with multiple scenarios constitute in this case as a 'single test'? Or is it each scenario inside a feature that would be the definition of a single test in this context?

I hope that makes sense,

Thank you for reading,

Zukky.

**Comment by rjanjua** [comment]
Hi Zukky,

3 scenarios in total, you can split them up into different feature files if you think that provides more clarity of your intent.

Regards

Rohan

**Comment by rjanjua** [comment]
I would say nice to meet you too but we've already met....

**Comment by ZukkyBaig** [comment]
Thanks Rohan, indeed we have. I hope Ramadhan is treating you well.

I have another question I hope you don't mind answering. In a typical BDD environment, you would create the feature files on input from a Business Analyst/Product Owner (if we are taking Agile roles into consideration), From this, usually small and bite-size feature files are created, which then background code can expand on further and add further complexity to (multiple commands for a single Gherkin line for example). How are you expecting this to be like? Step by step - (i.e. Given postcode is '<postcode>', And I click on British Gas as my electricity supplier, And I click on other supplier dropdown etc..) or small and clear like a BA/PO would define and read it i.e. Given I am on the supplier page, And I want compare Gas & Electricity, And I enter my Gas and Electricity details etc..Is the onus on me to choose which one or is this something that you guys can clarify?

**Comment by ZukkyBaig** [comment]
TL;DR -> Feature file(s) written from users perspective (high-level) or developers perspective (low-level)?

**Comment by rjanjua** [comment]
Hope your Ramadan is going well too.

The onus is on you to decide this. You'll have to justify your decision at the interview.

**Comment by ZukkyBaig** [comment]
Got it.

Thank you.
